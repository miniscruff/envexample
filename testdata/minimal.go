package testdata

//go:generate go run ../main.go -package testdata -struct MinimalConfig -export minimal.golden

// StartingConfig handles a configuration for a project with default env options.
type MinimalConfig struct {
	// Listen address for our http server that is configured
	// on multiple lines for testing.
	Address string `env:"ADDRESS" envDefault:"localhost"`
	// Listen port for our http server.
	Port int `env:"PORT" envDefault:"8080"`
}

/*
	Home         string        `env:"HOME"`
	Port         int           `env:"PORT" envDefault:"3000"`
	Password     string        `env:"PASSWORD,unset"`
	IsProduction bool          `env:"PRODUCTION"`
	Hosts        []string      `env:"HOSTS" envSeparator:":"`
	Duration     time.Duration `env:"DURATION"`
	TempFolder   string        `env:"TEMP_FOLDER" envDefault:"${HOME}/tmp" envExpand:"true"`
*/
