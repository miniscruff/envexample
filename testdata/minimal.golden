# Generated by envexample dev.

# StartingConfig handles a configuration for a project with default env options.
#
# Listen address for our http server that is configured
# on multiple lines for testing.
ADDRESS=localhost
# Listen port for our http server.
PORT=8080
